Dealer.h

```diff
@@ -0,0 +1,47 @@
+#ifndef DEALER_H
+#define DEALER_H
+#include "Deck.h"
+#include <iostream>
+#include <vector>
+#include <string>
+#include <string_view>
+class Dealer
+{
+private:
+  Deck deck{};
+  std::string m_name{};
+  std::vector<std::string> Hand{};
+  int m_score{};
+  static const int c_maxScore{21};
+public:
+  Dealer(std::string_view name)
+    : m_name{name}
+  {
+  }
+
+  friend std::ostream& operator<<(std::ostream& out, const Dealer& d)
+  {
+    out << "Dealer " << d.m_name << " has a score of << " << d.m_score << "\nHas cards: \n";
+    int it{};
+    for(const auto& cardStrings : d.Hand)
```

Comment:
'East const' is more in style these days than 'West const'. What I mean by that is you can do what you've done here 

`const auto &`

This is referred to as 'East const'. Or you can do 

`auto const &`

Which is referred to as 'West const'. They both mean the same thing, the reason for the preference has to do with pointers. Let's look at this line of code as an example

const auto *

In this case, what is the thing that's const? The pointer, or the thing that's pointed to? It's not really obvious. _But_ when you use 'West const' there is a natural way to read the const(s). Basically you read from right to left and the const always applies to the thing to the left of it. See the examples below

`auto const *` <- pointer to a constant type (eg what's pointed to is constant but the pointer can be changed to point to something else)

`auto * const` <- constant pointer to a mutable type (eg,. the pointer can't be changed to point to anything else but the value that's pointed at can be modified)

`auto const * const` <- a constant pointer to a constant type, neither the pointer nor pointed to value can be modified

Reply:
Nice. Your explanation clicked